app/src/main/java/org/cru/godtools/util/ActivityUtils.kt

```diff
@@ -0,0 +1,20 @@
+package org.cru.godtools.util
+
+import android.app.Activity
+import org.cru.godtools.article.activity.startCategoriesActivity
+import org.cru.godtools.model.Tool.Type
+import org.cru.godtools.tract.activity.TractActivity
+import org.cru.godtools.xml.service.ManifestManager
+import java.util.Locale
+
+fun openToolActivity(context: Activity, code: String, type: Type, vararg languages: Locale) {
+    // launch activity based on the tool type
+    when (type) {
+        Type.TRACT -> {
+            // start pre-loading the tract in the first language
+            ManifestManager.getInstance(context).getLatestPublishedManifest(code, languages[0])
+            TractActivity.start(context, code, *languages)
+        }
+        Type.ARTICLE -> context.startCategoriesActivity(code, languages[0])
+    }
+}
```

Comment:
How much does this project care about newline at the end of files?

Reply:
I personally try to enforce a new line on the last line. I've enabled some ktlint/checkstyle rules to enforce it in certain places, but not everywhere evidently